js/index.js

```diff
@@ -1,31 +1,37 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
-
-  //... your code goes here
+  let price = product.querySelector(".price span").innerHTML;
+  let quantity = product.querySelector(".quantity input").value;
+  let priceSubtotal = parseInt(price, 10) * parseInt(quantity, 10);
+  product.querySelector(".subtotal span").innerHTML = `${priceSubtotal}`;
+  return priceSubtotal;
 }
 
 function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
-
   // ITERATION 2
-  //... your code goes here
+
+  const products = document.querySelectorAll(".product");
+  let sum = [];
+  products.forEach((item) => {
+    sum.push(updateSubtotal(item));
+  });
 
   // ITERATION 3
-  //... your code goes here
+  document.querySelector("#total-value span").innerHTML = sum.reduce(
+    (a, b) => a + b
+  );
```

Comment:
If you made the sum a number, instead of an array, you could save yourself some time on the reduce.